packages/gatsby/src/utils/webpack.config.js

```diff
@@ -783,8 +783,8 @@ module.exports = async (
   }
 
   if (
-    process.env.GATSBY_EXPERIMENTAL_PRESERVE_WEBPACK_CACHE &&
-    (stage === `build-javascript` || stage === `build-html`)
+    process.env.GATSBY_EXPERIMENTAL_DEV_WEBPACK_CACHE &&
+    (stage === `develop` || stage === `develop-html`)
```

Comment:
But doesn't that disable it for the `gatsby build` step again?

Reply:
umm yes 🤦 

Clearly didn't spend very much time on this PR haha. Will push fix.